cloudpathlib/patches.py

```diff
@@ -0,0 +1,210 @@
+import builtins
+import os
+import os.path
+
+from .cloudpath import CloudPath
+
+
+def _check_first_arg(*args, **kwargs):
+    return isinstance(args[0], CloudPath)
+
+
+def _check_first_arg_first_index(*args, **kwargs):
+    return isinstance(args[0][0], CloudPath)
+
+
+def _patch_factory(original_version, cpl_version, cpl_check=_check_first_arg):
+    _original = original_version
+
+    def _patched_version(*args, **kwargs):
+        if cpl_check(*args, **kwargs):
+            return cpl_version(*args, **kwargs)
+        else:
+            return _original(*args, **kwargs)
+
+    original_version = _patched_version
+    return _patched_version
+
+
+def patch_open():
+    patched = _patch_factory(
+        builtins.open,
+        CloudPath.open,
+    )
+    builtins.open = patched
+    CloudPath.__fspath__ = lambda x: x  # turn off `fspath`
```

Comment:
So, the idea here is that even if code uses `fspath` to try to get the local path, we give back the `CloudPath` instead so it works downstream. This may break code that assumes `fspath` returns a string, which is what it is supposed to do.